packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -240,3 +87,167 @@ export abstract class S3BucketOrigin extends cloudfront.OriginBase {
     return { originAccessIdentity: '' };
   }
 }
+
+export class S3BucketOriginWithOAC extends S3BucketOrigin {
```

Comment:
Let's keep this class private if possible so we do not expose more than needed. I am worried that if we expose this then changing the API of this class will become breaking change.

Same goes to the `S3BucketOriginWithOAI` class below

Reply:
Yes these should definitely be private, thanks!